### DISCUSSION TITLE
How to install with vanill javascript?

### DISCUSSION THREAD

**Discussion by kars-h** [ueberdosis/tiptap#2947] [General Questions]
I am struggling to install tiptap with vanilla javascript.

The instructions (here: https://tiptap.dev/installation ) are saying that I can initialize the editor using:

```
import { Editor } from '@tiptap/core'
import StarterKit from '@tiptap/starter-kit'

new Editor({
  element: document.querySelector('.element'),
  extensions: [
    StarterKit,
  ],
  content: '<p>Hello World!</p>',
})
```

However, this doesn't work. I tried to move the @tiptap folder into a place the browser can reach.
I changed the code to:
```
<script type="module">
import { Editor } from '../node_modules/@tiptap/core'
import StarterKit from '../node_modules/@tiptap/starter-kit'

new Editor({
element: document.querySelector('.element'),
extensions: [
StarterKit,
],
content: '<p>Hello World!</p>',
})
</script>
```

 Then it anyway throws a 404 error:
`GET http://localhost:8100/node_modules/@tiptap/core/ net::ERR_ABORTED 404 (Not Found)`
`GET http://localhost:8100/node_modules/@tiptap/starter-kit/ net::ERR_ABORTED 404 (Not Found)`

I think I need to use webpack or browserify (StackOverflow tells me) to build the files first.
I've been failing with browserify and am getting frustrated because I must be oblivious to some essential webdev knowledge that tI don't seem to have. Because I can't find anyone else struggling with this...

I'm used to using a CDN or just put a .js file on my server and include that in a `<script src="/code.js">`

What am I missing? How can I get simply get the code running...

PS
I've found this CDN page https://tiptap.dev/installation/cdn and tried this, but that throws: `prosemirror-state.js:1 Uncaught RangeError: Adding different instances of a keyed plugin (plugin$)`

**Comment by sereneinserenade** [comment]
> I'm used to using a CDN or just put a .js file on my server and include that in a `<script src="/code.js"`>

At first thought, I think you're missing the `type="module"` in the script tag. take a look here, I'm using the CDN, and it works fine. https://github.com/django-tiptap/django_tiptap/blob/main/django_tiptap/templates/forms/tiptap_textarea.html#L357

**Comment by gr8tushar** [comment]
@kars-h You guessed it right. You'll have to use webpack or something similar to build it. They are node modules. They won't work out of the box in the browser.
The CDN files are actual js files which will work directly from the browser. 

The problem is that all the examples given are in type script, so if you use CDN, trying new things might be little difficult for you.

**Comment by kars-h** [comment]
Thanks for the replies.

I'm also not familiar with Typescript so perhaps that is indeed the problem.
I've gone with TinyMCE for now, and might give it another try in the future with webpack.
